pkgs/development/python-modules/dlib/default.nix

```diff
@@ -1,12 +1,17 @@
-{ buildPythonPackage, dlib, python, pytest }:
+{ buildPythonPackage, dlib, python, pytest, avxSupport ? true }:
 
 buildPythonPackage {
   inherit (dlib) name src nativeBuildInputs buildInputs meta;
 
+  # although AVX can be enabled, we never test with it. Some Hydra machines
+  # fail because of this, however their build results are probably used on hardware
+  # with AVX support.
   checkPhase = ''
     ${python.interpreter} nix_run_setup test --no USE_AVX_INSTRUCTIONS
```

Comment:
This should follow how it is built.

Reply:
I'm afraid I have to disagree here: as explained in the commit message and the comment above, some Hydra builders are affected by the problem as well. However their build results are likely used by machines that can (and hence should) use this feature.

Right now we ensure in every environment that the package is functional and we set a sensible default for our end users.

Reply:
We need to require an AVX-enabled bulder to build AVX-enabled software (and the same for all other CPU features)

Although GCC does not need that, there are tests like the one above (and there no common ```--no USE_AVX_INSTRUCTIONS``` to cover all the affected software; and the features like BMI/BMI2 affect all the packages) and the binaries compiled and run on ```configurePhase```.

Reply:
May I ask how you'd implement this currently? If I understand the discourse thread you've linked correctly, precise matching of optimization flags isn't implemented yet.

Reply:
> precise matching of optimization flags isn't implemented yet.

yes, so so far it is not implemented, I agree with you that test should use ```nix_run_setup test --no USE_AVX_INSTRUCTIONS``` because it is likely be executed  on the builder which does not not support AVX.
But once this will be implemented (and derivation's ```requiredSystemFeatures``` with require an AVX-enabled builder using the same feature name as in ```system-features``` in /etc/nix/nix.conf of the builder), ```--no USE_AVX_INSTRUCTIONS``` will no longer be needed

Reply:
full ack then. I'll resolve this discussion for now and subscribe myself to the discourse thread. As soon as we have a solution for this in nixpkgs, I can improve the `dlib` package. For now I'd like to see this merged and backported into the release.

Reply:
Currenly I am using names like ```gccarch-sandybridge```, ```gccarch-skylake-avx512``` but thay are difficult to compare (and to deduce that "skylake can build for sandybridge" but not vice versa)
So ```has-avx```, ```has-avx2```, ```has-avx512```, ```has-bmi```, ```has-aes``` might serve better as "system features".